picolog/picolog.c

```diff
@@ -0,0 +1,113 @@
+/*
+* Author: Christian Huitema
+* Copyright (c) 2019, Private Octopus, Inc.
+* All rights reserved.
+*
+* Permission to use, copy, modify, and distribute this software for any
+* purpose with or without fee is hereby granted, provided that the above
+* copyright notice and this permission notice appear in all copies.
+*
+* THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS" AND
+* ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE IMPLIED
+* WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE
+* DISCLAIMED. IN NO EVENT SHALL Private Octopus, Inc. BE LIABLE FOR ANY
+* DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES
+* (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES;
+* LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND
+* ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+* (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS
+* SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+*/
+#include <stdio.h>
+#include <stdlib.h>
+#include <string.h>
+#include <inttypes.h>
+
+#include "picoquic_internal.h"
+#include "txtlog.h"
+#ifdef _WINDOWS
+#include "../picoquicfirst/getopt.h"
+#endif
+
+FILE * open_outfile(const char * log_name, const char * out_file, const char * out_ext);
+
+void usage();
+
+int main(int argc, char ** argv)
+{
+    int ret = 0;
+
+    const char * log_name = NULL;
+    const char * out_format = "csv";
+    const char * out_file = NULL;
+
+    int opt;
+    while ((opt = getopt(argc, argv, "o:f")) != -1) {
+        switch (opt) {
+        case 'o':
+            out_file = optarg;
+            break;
+        case 'f':
+            out_format = optarg;
+            break;
```

Comment:
You may consider parsing the output format here, so you can reject incorrect output format early, and also provide "usage" feedback with the list of supported formats.